util/csvresponse.py

```diff
@@ -86,14 +88,32 @@ def _create_filter_list(self,keys):
         # start with fields we never want to include
         drop = {'id', 'annotations'}
         for key in keys:
-            # remove any "extra" keys while keeping qc params and removing 'provenance' params
-            if (key not in params_to_include and not self._is_qc_param(key)) or 'provenance' in key:
+            # remove any "extra" keys while keeping relevant qc params and removing 'provenance' params
+            if self._is_qc_param(key):
+                # only include if a requested param matches the QC Key
+                if not self._needs_qc_param(key,params_to_include):
```

Comment:
```key, params_to_include```

Reply:
Variable names, both here and in the next comment, are local to the functions and reflect the role in the function. I don't see any reason to change the variable names here.

Reply:
Just insert a space after the comma for PEP-8 compliance